node/src/components/linear_chain_sync/operations.rs

```diff
@@ -128,12 +195,42 @@ where
         {
             Ok(FetchedData::FromStorage { item }) => return Ok(Some(item)),
             Ok(FetchedData::FromPeer { item, .. }) => {
-                // TODO: Validate item, ban peer if necessary
                 // Compute the total weight of the validators
-                let _total_weight: U512 = trusted_validator_weights
-                    .iter()
-                    .map(|(_, weight)| *weight)
-                    .sum();
+                let BlockHeaderWithMetadata {
+                    block_header,
+                    block_signatures,
+                } = *item.clone();
+
+                match validate_finality_signatures(
+                    &block_header,
+                    &trusted_validator_weights,
+                    finality_threshold_fraction,
+                    &block_signatures,
+                ) {
+                    Ok(()) => {}
+                    Err(error) => {
+                        warn!(
+                            "Error validating finality signatures from peer. \
+                             peer: {peer:?}, \
+                             error: {error}",
+                            peer = peer.clone(),
+                            error = error
+                        );
```

Comment:
```suggestion
                        warn!(
                            ?peer, %error,
                            "Error validating finality signatures from peer.",
                        );
```

Reply:
Fixed in 98c5063df841b3bc09ac729c64b0770ca8ce67e4